internal/flavors/publisher_test.go

```diff
@@ -208,3 +210,140 @@ func sleepOnCycleEnd(t *testing.T, size int, eventCount int, interval time.Durat
+			expectedBufferCapacityOnEachPublish: []int{15},
+			expectedBufferCapacityEnd:           15,
+		},
+		"two batches under threshold": {
+			interval:                            2 * time.Second,
+			threshold:                           10,
+			incomeEventBatchesOf:                []int{5, 5},
+			expectedPublishedBatchesIDs:         [][]int{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
+			expectedBufferCapacityOnEachPublish: []int{15},
+			expectedBufferCapacityEnd:           15,
+		},
+		"two batches at threshold": {
+			interval:                            2 * time.Second,
+			threshold:                           5,
+			incomeEventBatchesOf:                []int{3, 3, 6},
+			expectedPublishedBatchesIDs:         [][]int{{1, 2, 3, 4, 5, 6}, {7, 8, 9, 10, 11, 12}},
+			expectedBufferCapacityOnEachPublish: []int{7, 7},
+			expectedBufferCapacityEnd:           7,
+		},
+		"single batch over threshold": {
+			interval:                            2 * time.Second,
+			threshold:                           3,
+			incomeEventBatchesOf:                []int{5},
+			expectedPublishedBatchesIDs:         [][]int{{1, 2, 3, 4, 5}},
+			expectedBufferCapacityOnEachPublish: []int{8},
+			expectedBufferCapacityEnd:           8, // 4 * 2
+		},
+		"single batch over threshold break capacity limit": {
+			interval:                            2 * time.Second,
+			threshold:                           3,
+			incomeEventBatchesOf:                []int{15},
+			expectedPublishedBatchesIDs:         [][]int{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}},
+			expectedBufferCapacityOnEachPublish: []int{15},
+			expectedBufferCapacityEnd:           4,
+		},
+		"multiple batches break capacity limit": {
+			interval:                            2 * time.Second,
+			threshold:                           5,
+			incomeEventBatchesOf:                []int{2, 2, 20},
+			expectedPublishedBatchesIDs:         [][]int{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24}},
+			expectedBufferCapacityOnEachPublish: []int{27},
+			expectedBufferCapacityEnd:           7,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			log := testhelper.NewLogger(t)
+			defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			var publisher *Publisher
+
+			client := newMockClient(t)
+			calls := make([]*mock.Call, len(tc.expectedPublishedBatchesIDs))
+			for i, publishedSliceIDs := range tc.expectedPublishedBatchesIDs {
+				calls[i] = client.EXPECT().PublishAll(events(publishedSliceIDs)).Run(
+					func(_ []beat.Event) {
+						assert.Equalf(t, tc.expectedBufferCapacityOnEachPublish[i], cap(publisher.eventsBuffer), "buffer capacity miss-match on publish number %d", i)
+					},
+				).Call
+			}
+			mock.InOrder(calls...)
+
+			publisher = NewPublisher(log, tc.interval, tc.threshold, client)
```

Comment:
```suggestion
			publisher = NewPublisher(log, time.Hour, tc.threshold, client)
```

Reply:
I think you suggest that because the channel is closed in the test sender? 

I was planning (and forgot) to make it part of the test case `closeChannel` as the current one. It gives better flexibility to the testcase rather than, essentially, disabling the interval at all. (I will add the `closeChannel` boolean to testcase struct).

Reply:
I am suggesting it in combination with the other suggestion in this review and the top-level comment:
> The interval doesn't play an role in this test so let's not mislead by putting it in the table